crypto/bn/bn_gcd.c

```diff
@@ -611,18 +612,30 @@ int BN_gcd(BIGNUM *r, const BIGNUM *in_a, const BIGNUM *in_b, BN_CTX *ctx)
         goto err;
 
     /* find shared powers of two, i.e. "shifts" >= 1 */
+    pow2_flag = 1;
+    pow2_chunk_index = 0;
+    pow2_shifts = 0;
     for (i = 0; i < r->dmax && i < g->dmax; i++) {
-        mask = ~(r->d[i] | g->d[i]);
-        for (j = 0; j < BN_BITS2; j++) {
-            bit &= mask;
-            shifts += bit;
-            mask >>= 1;
-        }
+        pow2_mask = r->d[i] | g->d[i];
+        pow2_temp = !constant_time_is_zero(pow2_flag) & !constant_time_is_zero_64(pow2_mask);;
+        pow2_flag &= !pow2_temp;
```

Comment:
pow2_flag value does not seem to be used?

Reply:
It is used in previous line.